README.md

```diff
@@ -92,7 +92,37 @@ A wildcard stands for all the non-specified HTTP methods.
 
 ##### Key: `<cache-header-name>` 
 
-A http cache header name (`Cache-control`, `Expires`, etc.).
+A http cache header name (`cache-control`, `expires`, `etag` etc.).
+
+###### ETags
+
+For Etags you can specify a custom generator in the configuration:
+
+```
+'etag' => array(
+    'override' => true,
+    'generator' => 'Your\Own\EtagGenerator'
+)
+```
+
+Example generator:
+
+```
+class EtagGenerator implements EtagGeneratorInterface
+{
+    /**
+     * Returns an ETag for the given response.
+     *
+     * @param HttpResponse $response
+     * @return string Etag
+     */
+    public function generateEtag(HttpResponse $response)
```

Comment:
One could want to create the Etag hash from the HTTP request (often simplier than generating the response to create the tag), thus, I would add the HTTP request too.